internal/store/migrate.go

```diff
@@ -0,0 +1,69 @@
+package store
+
+import (
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
+)
+
+// LatestVersion returns the version to which the last migration migrates.
+func LatestVersion() semver.Version {
+	return migrations[len(migrations)-1].toVersion
+}
+
+// Migrate advances the schema of the configured database to the latest version.
+func (sqlStore *SQLStore) Migrate() error {
+	var currentVersion semver.Version
+	if systemTableExists, err := sqlStore.tableExists("System"); err != nil {
+		return errors.Wrap(err, "failed to check if system table exists")
+	} else if systemTableExists {
+		currentVersion, err = sqlStore.getCurrentVersion(sqlStore.db)
+		if err != nil {
+			return err
+		}
+	}
+
+	sqlStore.logger.Infof(
+		"Schema version is %s, latest version is %s",
+		currentVersion,
+		LatestVersion(),
+	)
+
+	applied := 0
+	for _, migration := range migrations {
+		if !currentVersion.EQ(migration.fromVersion) {
+			break
+		}
+
+		sqlStore.logger.Infof("Migrating schema from %s to %s", currentVersion, migration.toVersion)
+		tx, err := sqlStore.db.Beginx()
+		if err != nil {
+			return errors.Wrapf(err, "failed to begin applying target version %s", migration.toVersion)
+		}
+		defer tx.Rollback()
```

Comment:
Does rollback not do anything if the commits are successful? Also what's the convention around defer in a loop? I feel like it would be better to wrap the contents of the loop in a function so that the defers can't pile up

Reply:
Yeah, `tx.Rollback` is a no-op if the transaction has already committed (or rolled back). This makes for a nice clean-up paradigm.

But you're 100% right on the defer in a loop -- that's just horribly wrong on my part. Will fix.